phlay

```diff
@@ -306,7 +306,15 @@ class Diff:
             b_body = check_output(['git', 'cat-file', 'blob', b_blob])
 
         # Detect if we're binary, and generate a unified diff
-        change.binary = b'\0' in a_body or b'\0' in b_body
+        def is_binary(body):
+            try:
+                body.decode('utf-8')
+            except UnicodeError:
+                return True
+            return b'\0' in body
```

Comment:
The `b'\0' in body` check is meant to be used as an early heuristic to avoid trying to decode the file if it's definitely binary. Can we do that check first, and only try to decode as utf-8 if it succeeds?
```suggestion
            if b'\0' in body:
                return True
            try:
                body.decode('utf-8')
            except UnicodeError:
                return True
            return False
```

Reply:
Yeah, that's a better idea!